File: src/main/java/net/snipsniper/config/ConfigOption.java
package net.snipsniper.config;

import org.capturecoop.cclogger.CCLogger;
import org.capturecoop.cclogger.CCLogLevel;

public class ConfigOption {
    private String key;
    private String value;
    private final TYPE type;

    public ConfigOption(String key, String value) {
        this.key = key;
        this.value = value;
        type = TYPE.KEY_VALUE;
    }

    public ConfigOption(String comment) {
        if(comment.startsWith("#"))
            value = comment.substring(1);
        else
            value = comment; //Avoid reference
        type = TYPE.COMMENT;
    }

    public ConfigOption() {
        type = TYPE.NEWLINE;
    }

    public void setValue(String value) {
        this.value = value + "";
    }

    public String getKey() {
        return key;
    }

    public String getValue() {
        return value;
    }

    public String toString() {
        switch(type) {
            case KEY_VALUE: return key + "=" + value;
            case COMMENT: return "#" + value;
            case NEWLINE: return "";
        }
        CCLogger.log("NO TYPE SET! KEY: %c VALUE: %c", CCLogLevel.ERROR, key, value);
        CCLogger.logStacktrace(CCLogLevel.ERROR);
        return null;
    }

    public TYPE getType() {
        return type;
    }

    enum TYPE { KEY_VALUE, COMMENT, NEWLINE}
}

File: src/main/kotlin/net/snipsniper/config/ConfigOption.kt
package net.snipsniper.config

class ConfigOption {
    enum class TYPE {KEY_VALUE, COMMENT, NEWLINE}
    val key: String
    var value: String
    val type: TYPE

    constructor(key: String, value: String) {
        this.key = key
        this.value = value
        type = TYPE.KEY_VALUE
    }

    constructor(comment: String) {
        key = ""; value = comment
        type = TYPE.COMMENT
    }

    constructor() {
        key = ""; value = ""
        type = TYPE.NEWLINE
    }

    override fun toString(): String {
        return when (type) {
            TYPE.KEY_VALUE -> "$key=$value"
            TYPE.COMMENT -> "#$value"
            TYPE.NEWLINE -> ""
        }
    }
}